packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2584,15 +2584,13 @@ export class StripeHelper {
       total: invoiceTotalInCents,
       subtotal: invoiceSubtotalInCents,
       hosted_invoice_url: invoiceLink,
-      lines: {
-        data: [
-          {
-            period: { end: nextInvoiceDate },
-          },
-        ],
-      },
+      lines: { data: invoiceLines },
     } = invoice;
 
+    const nextInvoiceDate =
+      invoiceLines.find((line) => line.type === 'subscription')?.period.end ||
+      invoiceLines[0].period.end;
```

Comment:
I don't think we'd ever want to use a non-`subscription` type `listitem` as the next invoice date, would we? If none of the `listitem`s are of type `subscription`, perhaps we should throw an error, just as we do if there's no plan found above?

It'd be a good idea to add a test case for this error scenario as well.

Reply:
No I don't think we'd ever use a non-`subscription` type. I just wasn't exactly sure what the `invoice` line items would be available, so I left the default to be the logic we had before this change.

After doing a bit more reading [in the Stripe docs](https://stripe.com/docs/billing/invoices/subscription#adding-upcoming-invoice-items) and on their discord, I believe for invoices of a subscription, the `subscription` type should always be available.

I've updated the logic to throw an error, and added a test for it as well.